pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/IndexService.java

```diff
@@ -54,7 +56,7 @@ public class IndexService {
 
   private static volatile IndexService _instance = fromServiceLoader();
 
-  private final Set<IndexType<?, ?, ?>> _allIndexes;
+  private final List<IndexType<?, ?, ?>> _allIndexes;
   private final Map<String, IndexType<?, ?, ?>> _allIndexesById;
```

Comment:
Consider directly creating an `ImmutableSortedMap` instead

Reply:
I assume SortedMap comes with some extra cost when doing lookup, and lookup on this map happens a lot. Instead, the sorting cost just gets paid once when initializing.

Reply:
found that sortedMap lookup could be 2.5X slower than hashMap, with a simple benchmark like below. So will keep using the ImmutableMap for here.

```
Benchmark                   Mode  Cnt   Score   Error  Units
BenchmarkMap.basicMap       avgt   10  29.088 ± 0.446  us/op
BenchmarkMap.basicMap:jfr   avgt          NaN            ---
BenchmarkMap.sortedMap      avgt   10  73.264 ± 1.637  us/op
BenchmarkMap.sortedMap:jfr  avgt          NaN            ---

// _basicMap, _sortedMap, and _indexesAry contained all 10+ index types
...
  @Benchmark
  public void basicMap(Blackhole blackhole) {
    for (int i = 0; i < 10_000; i++) {
      IndexType<?, ?, ?> idx = _basicMap.get(_indexesAry[i % _indexesAry.length]);
      blackhole.consume(idx);
    }
  }

  @Benchmark
  public void sortedMap(Blackhole blackhole) {
    for (int i = 0; i < 10_000; i++) {
      IndexType<?, ?, ?> idx = _sortedMap.get(_indexesAry[i % _indexesAry.length]);
      blackhole.consume(idx);
    }
  }
}
```